aligner_seed.cpp

```diff
@@ -749,43 +718,49 @@ inline bool startSearchSeedBi(
 	const TIndexOffU *ftab,
 	const TIndexOffU *eftab,
 	const TIndexOffU fchr[5],     // index fchr
-	const CacheAndSeed &pcs,
-	SeedAlignerSearchState &sstate,
-	BwtTopBotFw &bwt)
+	const SeedAlignerSearchData &sdata,
+	SeedAlignerSearchState &sstate)
 {
 
+	BwtTopBotFw bwt = sdata.bwt
 	assert_eq(sstate.step, 0);
-	assert_gt(pcs.n_seed_steps(), 0);
+	assert_gt(sdata.n_seed_steps(), 0);
 	{
-		const char *seq = pcs.seq;
 		// Just starting
 		assert(!sstate.tloc.valid());
 		assert(!sstate.bloc.valid());
-		const int seed_step_min = pcs.seed_step_min();
+		const int seed_step_min = sdata.seed_step_min();
 		int off = abs(seed_step_min)-1;
 		// Check whether/how far we can jump using ftab or fchr
 		int ftabLen = ep.ftabChars();
-		if (ftabLen > 1 && ftabLen <= pcs.maxjump()) {
-			TIndexOffU fwi0 = Ebwt::ftabSeqToInt(ftabLen, true, seq, off - ftabLen + 1, false);
+		if (ftabLen > 1 && ftabLen <= sdata.maxjump()) {
+
+			// Section below is equivalent to Ebwt::ftabSeqToInt(ftabLen, true, seq, off - ftabLen + 1, false);
+			uint64_t low_bits = sdata.sak.seq >> (2 * (sdata.sak.len - 1 - off));
```

Comment:
Could you implement this as a function, instead of inlining?
E.g. you could call it
Ebwt::ftabSakToInt